src/dc_receive_imf.rs

```diff
@@ -1765,8 +1766,15 @@ async fn create_or_lookup_mailinglist(
     }
 
     // as a last resort, use the ListId as the name
+    // but strip some known, long hash prefixes
     if name.is_empty() {
         name = listid.clone();
+
+        static PREFIX_32_CHARS_HEX: Lazy<Regex> =
+            Lazy::new(|| Regex::new("([0-9a-fA-F]{32})\\.(.{6,})").unwrap());
+        if let Some(cap) = PREFIX_32_CHARS_HEX.captures(&name) {
+            name = cap[2].to_string();
+        }
```

Comment:
```suggestion
    if name.is_empty() {
        static PREFIX_32_CHARS_HEX: Lazy<Regex> =
            Lazy::new(|| Regex::new(r"([0-9a-fA-F]{32})\.(.{6,})").unwrap());
        if let Some(cap) = PREFIX_32_CHARS_HEX.captures(&listid) {
            name = cap[2].to_string();
        } else {
            name = listid.clone();
        }
```

Note that this is untested, maybe there are some problems with this, and also maybe that's opinion based, but:
- Don't mutate name, but just set it once
- Use a raw string: `r"([0-9a-fA-F]{32})\.(.{6,})"` to avoid the `\\.`

Reply:
although `name` is mutated anyway, your suggestion is still better readable, thanks a lot!